### DISCUSSION TITLE
Connecting a USB hub to the USB OTG port on the Icicle kit

### DISCUSSION THREAD

**Discussion by aeb190** [polarfire-soc/polarfire-soc-documentation#514] [How to]
I've experienced some trouble in trying to connect multiple devices to the Icicle kit via an external USB hub. I was able to connect a single USB keyboard to the USB OTG port via a USB A to USB micro b adapter. However, the board only seems to recognize one device if I try to connect multiple devices to an external hub. Does anyone have experience with this issue? Ubuntu is currently installed on the board.

**Comment by hughbreslin** [comment]
Hey @aeb190 are you using a powered external hub or just a standard one? The kit can only provide so much power over the microusb port which may be causing your issue depending on what you're connecting up
